pov_tools/create_POV_data_files

```diff
@@ -0,0 +1,201 @@
+#!/bin/bash
+
+# create_POV_data_files
+# version - all .png files that don't contain 'preview' are processed
+OIFS="$IFS"
+IFS=$'\n'
+cd "$(dirname "$0")"
+clear
+height_only=0
+preset=0
+download=0
+
+echo ""
+echo "Welcome! Let's make some POV images!"
+echo ""
+echo "Found these image files:"
+# get .png files except containing 'preview' and build array
+pngs=($(find . -maxdepth 1 -type f -name "*.png" ! -name "*preview*"))
+for strip in ${pngs[@]}; do echo "${strip##*/}"; done
+# get number of files in array
+files=${#pngs[@]}
+echo ""
+
+# If NO images in the folder, offer logo
+if [[ $files -lt 1 ]]; then
+	echo ""
+	echo "OOPS! No .png image files found!"
+	echo ""
+	echo "Do you want to download a sample Star Wars logo image?"
+	echo "Type y for yes,"
+	echo "type q to quit,"
+	read -p $'or press ENTER to start over..\n' dl
+	case $dl in
+		y) download=1;;
+		q) exit;;
+		*) exec "./create_POV_data_files";;
+	esac
+fi
+if [[ $download == 1 ]]; then
+	echo ""
+	wget https://upload.wikimedia.org/wikipedia/commons/thumb/6/6c/Star_Wars_Logo.svg/1024px-Star_Wars_Logo.svg.png
+	pngs=$"1024px-Star_Wars_Logo.svg.png"
+fi
+
+valid_option=0
+while [[ $valid_option != 1 ]]; do
+	echo "Please select the settings you want to create the POV image(s) with:"
+	echo "	height - number of pixels in your blade (one strip emitter to tip)"
+	echo "	length - blade length in the measurement unit of choice"
+	echo "	offset - the distance (in same unit as length) from the bottom of the blade"
+	echo "		 to the fulcrum (e.g. the joint you are swinging with)"
+	echo ""
+	echo "1 - Preset 1 (units=inches height=144 length=39 offset=18)"
+	echo "2 - Preset 2 (units=centimeters height=97 length=63 offset=50)"
+	echo "3 - Preset 3 (units=centimeters height=72 length=50 offset=15)"
+	echo "4 - Preset 4 (units=centimeters height=72 length=50 offset=39)"
+	echo "5 - Height entry only"
+	echo "6 - Custom options"
+	echo "q - to quit"
+	read option
+	case $option in
+		"") echo "! Some option is needed. Please try again"; echo "";;
+		1) units="inches" height=144 length=39 offset=18 preset=1 valid_option=1;;
+		2) units="centimeters" height=97 length=63 offset=50 preset=1 valid_option=1;;
+		3) units="centimeters" height=72 length=50 offset=15 preset=1 valid_option=1;;
+		4) units="centimeters" height=72 length=50 offset=39 preset=1 valid_option=1;;
+		5) height_only=1 valid_option=1;;
+		6) valid_option=1;;
```

Comment:
"custom options" (6) doesn't seem to do anything, how is that supposed to work?

Reply:
With option 6 the whole package of questions is getting asked:
- how many pixels in blade
- preferred unit
- length of blade (in selected unit) (there is a default suggestion based on pixels and selected unit)
- offset (in selected unit) (there is a default option based on selected unit that matches the 18" from windshieldwiper default)
- -> make

Reply:
the valid_option=1 helps to break the while-do loop

Reply:
I get that option 6 exits the menu, but how does it make it "custom" ?

Reply:
Custom options are then input by the questionnaire that follows..